liberty-maven-plugin/src/main/java/io/openliberty/tools/maven/server/DevMojo.java

```diff
@@ -349,57 +463,64 @@ public boolean recompileBuildFile(File buildFile, List<String> artifactPaths, Th
+                            addArtifacts(root, artifactsList);
+                            for (File a : artifactsList) {
+                                log.debug("Artifact: " + a);
+                                if (a.getCanonicalPath().endsWith(".jar")) {
+                                    addToClassPath.add(a.getCanonicalPath());
                                 }
-                            } catch (DependencyResolutionException e) {
-                                throw new MojoExecutionException(e.getMessage(), e);
                             }
                             artifactPaths.addAll(addToClassPath);
                         }
+                    }
+                }
 
-                        if (!dependencyIds.isEmpty()) {
-                            runLibertyMojoInstallFeature(null);
-                            dependencyIds.clear();
-                        }
-
-                        // update dependencies
-                        for (Artifact artifact : updatedArtifacts) {
-                            for (Dependency dependency : dependencies) {
-                                if (artifact.getArtifactId().equals(dependency.getArtifactId())) {
-                                    this.existingDependencies.add(dependency);
-                                    break;
-                                }
-                            }
-                        }
-                        if (!isUsingBoost() && unhandledChange) {
-                            log.warn(
-                                    "Unhandled change detected in pom.xml. Restart liberty:dev mode for it to take effect.");
-                        } else {
-                            this.existingPom = modifiedPom;
-                        }
-                        return true;
-                    } else if (isUsingBoost()) {
-                        this.existingPom = modifiedPom;
-                        return true;
-                    } else if (unhandledChange) {
-                        log.warn(
-                                "Unhandled change detected in pom.xml. Restart liberty:dev mode for it to take effect.");
+                if (restartServer) {
+                    // TODO: restart server automatically
+                    // - stop Server
+                    // - create server or runBootMojo
+                    // - install feature
+                    // - deploy app
+                    // - start server
+                    log.error("Changes for Liberty server bootstrap properties or environment variables or JVM options "
+                            + "in the pom.xml has been detected, Restart liberty:dev mode for it to take effect.");
+                    project = backupProject;
+                    session.setCurrentProject(backupProject);
+                    return false;
+                } else {
+                    if (isUsingBoost() && (createServer || runBoostPackage)) {
```

Comment:
Since this is in the `else` statement, if `restartServer` is true above, then boost:package will never run.  Is this intended?

Reply:
Yes, this is intended.  boost:package will run during restart server if this is a boost project (see the TODO comment)

```
if (restartServer) {
                    // TODO: restart server automatically
                    // - stop Server
                    // - create server or runBootMojo
                    // - install feature
                    // - deploy app
                    // - start server
```

Reply:
But for now since it just prints a message instead of restarting, if it's a boost project, it would be better if that runs boost package

Reply:
boost package can only update the server configuration with the new bootstrap properties and server env, but it can't restart server.  If we design to call boost package, we should also call liberty:create to update the target server configuration.  Also set the latest project obj to current Session.  We probably don't need to run and test (i.e return false).

Is that what we want for now until we have the server restart capability?

```
               if (restartServer) {
                    // TODO: restart server automatically
                    // - stop Server
                    // - create server or runBootMojo
                    // - install feature
                    // - deploy app
                    // - start server
                    if (isUsingBoost) {
                        runBoostMojo("package", false);
                    } else {
                        runLibertyMojoCreate();
                    }
                    log.error("Changes for Liberty server bootstrap properties, environment variables or JVM options "
                            + "in the pom.xml have been detected. Restart liberty:dev mode for the changes to take effect.");
                    //project = backupProject;
                    //session.setCurrentProject(backupProject);
                    return false;
                } else {
```

Reply:
In that case, I think we should leave it as is for now.